bin/main.ml

```diff
@@ -809,6 +809,44 @@ let utop =
   in
   (term, Term.info "utop" ~doc ~man )
 
+module Init = struct
+  open Init
+  let lib =
+    let doc = "The project is intended to be a library" in
+    (Kind.Library, Arg.info ["lib"; "library"] ~doc)
+
+  let bin =
```

Comment:
I don't think that these will be the only types of projects, so I suggest refactoring that into `--kind lib` / `--kind bin`. This also removes the weird quirk about what happens when you pass `--lib --bin` (it seems that all are ignored except the last one). What do you think about this?

Reply:
I realize that this does not go in the same direction as what @rgrinberg suggested below - the point he makes about having specific docs & flags is a good one, not sure how to reconcile both.

Reply:
Separate sub commands for every kind are preferable as they will allow us to have more specific `--help` for every sub command as most of the kinds have specific options.

Reply:
And they prevent passing several kinds so I'm 👍 for that.

Reply:
I do think nested sub-commands generally make for nicer UI, so I am on board for this change.